src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -20,227 +129,2330 @@ const ActorInit En_Ossan_InitVars = {
+                    }
+                }
+            } else {
+                phi_v0 = temp_v0 + 2;
+                if (phi_v0 >= 8) {
+                    phi_v0 = 5;
+                }
+                while (phi_v0 != temp_v0) {
+                    if (this->unk_200[phi_v0] != NULL) {
+                        this->unk_252 = phi_v0;
+                        return;
+                    }
+                    phi_v0 += 2;
+                    if (phi_v0 >= 8) {
+                        phi_v0 = 5;
+                    }
+                }
+            }
+        }
+    }
+}
+
+s32 func_80AC47DC(GlobalContext* globalCtx, EnOssan* this, Input* cont1) {
+    EnGirlA* girlA = this->unk_200[this->unk_252];
+
+    if (func_80AC3884(this, globalCtx, cont1)) {
+        return 1;
+    }
+    if (func_80106BC8(globalCtx)) {
+        if (!(girlA->actor.params == 38) && !girlA->unk_1A8) {
+            this->unk_1FE = this->unk_1FC;
+            func_8010B720(globalCtx, this->unk_200[this->unk_252]->unk_1A0);
+            this->unk_288 = 0;
+            this->unk_2C0 = 0;
+            switch (girlA->actor.params) {
+                case 30:
+                case 31:
+                case 32:
+                case 33:
+                case 34:
+                case 35:
+                case 36:
+                case 37:
+                    func_80078884(0x4808);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 0x18;
+                    return 1;
+                case 17:
+                    func_80078884(0x4808);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 0xA;
+                    return 1;
+                case 18:
+                    func_80078884(0x4808);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 0xB;
+                    return 1;
+                case 19:
+                case 20:
+                    func_80078884(0x4806);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 0xC;
+                    return 1;
+                case 3:
+                case 6:
+                case 45:
+                case 46:
+                case 47:
+                    func_80078884(0x4808);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 0xD;
+                    return 1;
+                default:
+                    func_80078884(0x4808);
+                    this->unk_251 = 0;
+                    this->unk_1FC = 9;
+                    return 1;
+            }
+        } else {
+            func_80078884(0x4806);
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void func_80AC4978(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    s32 pad;
+    s32 idx;
+    u8 tmp = this->unk_252;
+
+    if (!func_80AC652C(this)) {
+        // ZOOMING!
+        osSyncPrintf("%s[%d]:" VT_FGCOL(GREEN) "ズーム中！！" VT_RST "\n", "../z_en_oB1.c", 2152);
+        this->unk_1E6 = 3;
+        return;
```

Comment:
I think most of the returns in this function can be combined in to else / else if statements

Reply:
Remaining 2 required for matching.